src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs

```diff
@@ -31,6 +31,54 @@ public void ExtractXMLFromDocComment()
                 Assert.Equal(expectedXMLFragment, extractedXMLFragment);
             }
 
+            [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+            [WorkItem(42986, "https://github.com/dotnet/roslyn/issues/42986")]
+            public async Task TestNativeInteger()
+            {
+                var metadataSource = "public class C { public nint i; public nuint i2; }";
+                var symbolName = "C";
+
+                await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, languageVersion: "Preview", metadataLanguageVersion: "Preview",
+                    expected: $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+using System;
+using System.Runtime.CompilerServices;
+
+public class [|C|]
+{{
+    [NativeIntegerAttribute]
```

Comment:
We should be dropping these, right?

Reply:
I don't think so. Notice the test below for tuples which I included for reference.